langchain4j-oracle/src/test/java/dev/langchain4j/store/embedding/oracle/CommonTestOperations.java

```diff
@@ -0,0 +1,219 @@
+import dev.langchain4j.store.embedding.EmbeddingStore;
+import oracle.sql.CHAR;
+import oracle.sql.CharacterSet;
+import oracle.ucp.jdbc.PoolDataSource;
+import oracle.ucp.jdbc.PoolDataSourceFactory;
+import org.testcontainers.oracle.OracleContainer;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.logging.Logger;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * A collection of operations which are shared by tests in this package.
+ */
+final class CommonTestOperations {
+
+    /**
+     * Model used to generate embeddings for this test. The all-MiniLM-L6-v2 model is chosen for consistency with other
+     * implementations of EmbeddingStoreIT.
+     */
+    private static final EmbeddingModel EMBEDDING_MODEL = new AllMiniLmL6V2QuantizedEmbeddingModel();
+
+    /** Name of a database table used by tests */
+    public static final String TABLE_NAME = "LANGCHAIN4J_EMBEDDING_STORE";
+
+    /**
+     * Seed for random numbers. When a test fails, "-Ddev.langchain4j.store.embedding.oracle.SEED=..." can be used to
+     * re-execute it with the same random numbers.
+     */
+    private static final long SEED = Long.getLong(
+            "dev.langchain4j.store.embedding.oracle.SEED", System.currentTimeMillis());
+    static {
+        Logger.getLogger(CommonTestOperations.class.getName())
+                .info("dev.langchain4j.store.embedding.oracle.SEED=" + SEED);
+    }
+
+    /**
+     * Used to generate random numbers, such as those for an embedding vector.
+     */
+    private static final Random RANDOM = new Random(SEED);
+
+    private CommonTestOperations() {}
+
+    private static final PoolDataSource DATA_SOURCE = PoolDataSourceFactory.getPoolDataSource();
+
+    static {
+        try {
+            DATA_SOURCE.setConnectionFactoryClassName("oracle.jdbc.datasource.impl.OracleDataSource");
+            String urlFromEnv = System.getenv("ORACLE_JDBC_URL");
+
+            if (urlFromEnv == null) {
+                // The Ryuk component is relied upon to stop this container.
+                OracleContainer oracleContainer =
+                    new OracleContainerWithTimeout("gvenzl/oracle-free:23.4-slim-faststart")
+                        .withStartupTimeoutSeconds(600)
+                        .withConnectTimeoutSeconds(600)
+                        .withDatabaseName("pdb1")
+                        .withUsername("testuser")
+                        .withPassword("testpwd");
+                oracleContainer.start();
+
+                DATA_SOURCE.setURL(oracleContainer.getJdbcUrl());
+                DATA_SOURCE.setUser(oracleContainer.getUsername());
+                DATA_SOURCE.setPassword(oracleContainer.getPassword());
+            }
```

Comment:
There is no need to create `OracleContainerWithTimeout`. There is an issue with `withStartupTimeoutSeconds` that is tracked in https://github.com/testcontainers/testcontainers-java/issues/8590 but `withStartupTimeout(Duration.ofSeconds(600))` can be used.

```suggestion
                OracleContainer oracleContainer =
                    new OracleContainer("gvenzl/oracle-free:23.4-slim-faststart")
                        .withStartupTimeout(Duration.ofSeconds(600))
                        .withConnectTimeoutSeconds(600)
                        .withDatabaseName("pdb1")
                        .withUsername("testuser")
                        .withPassword("testpwd");
                oracleContainer.start();

                DATA_SOURCE.setURL(oracleContainer.getJdbcUrl());
                DATA_SOURCE.setUser(oracleContainer.getUsername());
                DATA_SOURCE.setPassword(oracleContainer.getPassword());
            }
```

Reply:
Thank you!

Reply:
That suggested changes are not enough, we should also remove the OracleContainerWithTimeout class.